domain/sysvar_cache.go

```diff
@@ -149,6 +149,9 @@ func (do *Domain) rebuildSysVarCache(ctx sessionctx.Context) error {
 				if err != nil {
 					logutil.BgLogger().Error(fmt.Sprintf("load global variable %s error", sv.Name), zap.Error(err))
 				}
+				if sv.Name == variable.TiDBStatsCacheMemQuota {
+					do.SetStatsCacheCapacity(variable.StatsCacheMemQuota.Load())
+				}
```

Comment:
Maybe we can do it in `SetGlobal`?

Reply:
It's hard to get `domain` in `SetGlobal`.

Reply:
Morgan just did a refactor to remove the specific logic here... And I do like this way:
https://github.com/pingcap/tidb/pull/33054/files#diff-93901abecfb2298268c7204c8665c636fa37285005c6b53fd3a552cd163c3e7eR887
Maybe we can have a discussion with Morgan to have an alignment.

Reply:
Yeah, it's a known issue with getting the domain :( Unfortunately you have to do this trick in sysvar_cache.go.